### PR TITLE
[12.0][ADD] Add module custom_image

### PR DESCRIPTION
**Pull request by p-tombez**
Port of https://github.com/OCA/server-tools/pull/1377 in V12

### CODE REVIEW

File: base_conditional_image/models/image.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            tools.image_resize_images(vals)
+        elif required:
+            raise ValidationError(
+                _('At least one image type must be specified')
+            )
+
+    @api.model
+    def create(self, vals):
+        self._process_images(vals, required=True)
+        return super(Image, self).create(vals)
```
Comment (bealdav) [inline base_conditional_image/models/image.py]:
```suggestion
        return super().create(vals)
```
In python 3
https://docs.python.org/3/library/functions.html#super

File: base_conditional_image/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
```
Comment (bealdav) [inline base_conditional_image/__init__.py]:
```suggestion
```
Please remove these marker over python files : that's the default encoding

File: base_conditional_image/models/image.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import api, fields, models, tools, _
+from openerp.exceptions import ValidationError
```
Comment (bealdav) [inline base_conditional_image/models/image.py]:
```suggestion
from odoo.exceptions import ValidationError
```

All over files

File: base_conditional_image/models/abstract_image.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import fields, models
+from openerp.tools.safe_eval import safe_eval
+
+
+class AbstractImageStorage(models.AbstractModel):
+    _name = 'abstract.conditional.image'
```
Comment (bealdav) [inline base_conditional_image/models/abstract_image.py:9]: Is a there a reason why there is no matching between _name and class name ?
Not blocking

**Review by bealdav — CHANGES_REQUESTED** [review]
Some typo errors for this version

**Comment by p-tombez** [comment]
Completely missed these, thanks for the feedback

**Comment by bealdav** [comment]
Not mandatory but about adhoc repository, there is a new one since v10.
https://github.com/OCA/storage/tree/12.0
As your module brings a real added value on image management, maybe there is a better place than server-tools.
What do you think ? @sebastienbeau @rousseldenis

File: base_conditional_image/models/image.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+class Image(models.Model):
+    _name = 'image'
+
+    name = fields.Char('Name', required=True)
+    model_name = fields.Char('Model Name', required=True)
+    selector = fields.Text(
+        'Selector',
+        help='Python expression used as selector when multiple images are used'
+             'for the same object. The variable `object` refer '
```
Comment (asaunier) [inline base_conditional_image/models/image.py]:
```suggestion
             'for the same object. The variable `object` refers '
```

File: base_conditional_image/models/image.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+             'to the actual record on which the expression will be executed. '
+             'An empty expression will always return `True`.'
+    )
+    company_id = fields.Many2one(
+        'res.company', 'Company',
+        help='Company related check. If inherited object does not have a '
+             '`company_id` field, it will be ignored. '
+             'The check will first take the records with a company then, '
+             'if not match is found, the ones without a company.'
```
Comment (asaunier) [inline base_conditional_image/models/image.py]:
```suggestion
             'if no match is found, the ones without a company.'
```

**Review by asaunier — COMMENTED** [review]
Only minor typos, else LGTM

**Review by Tonow-c2c — APPROVED** [review]
Code LGTM

**Review by damdam-s — APPROVED** [review]
LGTM. code review only

**Comment by dreispt** [comment]
@p-tombez can you please incorporate the suggested fixes, so that we can proceed to a merge?
Thanks!

**Pull request merged**